tensorflow/core/kernels/data/map_dataset_op.cc

```diff
@@ -12,219 +12,194 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+          // that we should terminate the iteration early.
+          *end_of_sequence = true;
+          return Status::OK();
         }
+      } else {
+        return s;
       }
+    }
 
-     protected:
-      std::shared_ptr<model::Node> CreateNode(
-          IteratorContext* ctx, model::Node::Args args) const override {
-        return model::MakeKnownRatioNode(std::move(args),
-                                         /*ratio=*/1);
-      }
-
-      Status SaveInternal(IteratorStateWriter* writer) override {
-        TF_RETURN_IF_ERROR(SaveInput(writer, input_impl_));
-        return Status::OK();
-      }
+   protected:
+    std::shared_ptr<model::Node> CreateNode(
+        IteratorContext* ctx, model::Node::Args args) const override {
+      return model::MakeKnownRatioNode(std::move(args), /*ratio=*/1);
+    }
 
-      Status RestoreInternal(IteratorContext* ctx,
-                             IteratorStateReader* reader) override {
-        TF_RETURN_IF_ERROR(RestoreInput(ctx, reader, input_impl_));
-        return Status::OK();
-      }
+    Status SaveInternal(IteratorStateWriter* writer) override {
+      TF_RETURN_IF_ERROR(SaveInput(writer, input_impl_));
+      return Status::OK();
+    }
 
-     private:
-      std::unique_ptr<IteratorBase> input_impl_;
-      std::unique_ptr<InstantiatedCapturedFunction> instantiated_captured_func_;
-    };
+    Status RestoreInternal(IteratorContext* ctx,
+                           IteratorStateReader* reader) override {
+      TF_RETURN_IF_ERROR(RestoreInput(ctx, reader, input_impl_));
+      return Status::OK();
+    }
 
-    const DatasetBase* const input_;
-    const bool preserve_cardinality_;
-    const std::unique_ptr<CapturedFunction> captured_func_;
-    const DataTypeVector output_types_;
-    const std::vector<PartialTensorShape> output_shapes_;
+   private:
+    std::unique_ptr<IteratorBase> input_impl_;
+    std::unique_ptr<InstantiatedCapturedFunction> instantiated_captured_func_;
   };
 
-  std::shared_ptr<FunctionMetadata> func_metadata_ = nullptr;
-  DataTypeVector output_types_;
-  std::vector<PartialTensorShape> output_shapes_;
-  bool preserve_cardinality_;
+  const DatasetBase* const input_;
+  const bool preserve_cardinality_;
+  const std::unique_ptr<CapturedFunction> captured_func_;
+  const DataTypeVector output_types_;
+  const std::vector<PartialTensorShape> output_shapes_;
 };
 
+void MapDatasetOp::MakeDataset(OpKernelContext* ctx, DatasetBase* input,
+                               DatasetBase** output) {
+  std::unique_ptr<CapturedFunction> captured_func;
+  OP_REQUIRES_OK(
+      ctx, CapturedFunction::Create(ctx, func_metadata_, "other_arguments",
+                                    &captured_func));
+
+  *output = new Dataset(ctx, input, std::move(captured_func), output_types_,
+                        output_shapes_, preserve_cardinality_);
+}
+
 REGISTER_KERNEL_BUILDER(Name("MapDataset").Device(DEVICE_CPU), MapDatasetOp);
```

Comment:
put the kernel registration into anonymous namespace ... applies to other .cc files as well